opendatahub-io/odh-dashboard
# Move admin and allowed groups to OdhDashboardConfig

Resolves: #383 

Moves admin and allowed groups to the CR.

## Description
Adds a new field in the OdhDashboardConfig called groupsConfig. Removes logic dealing with configmaps and gets groups directly based on what is in the CR. Replaces logic and types to accomodate this change.

The PR additionally disables the notebook controller page if the user is not in the allowed groups to resolve https://github.com/opendatahub-io/odh-dashboard/issues/342

## How Has This Been Tested?
1. Modify adminGroups in the groupsConfig section of the CR
2. Verify relevant admin permissions are applied when accesing the dashboard route.
3. Modify allowedGroups.
4. Verify access to the notebook controller

## Merge criteria:
